rustsec/src/advisory/license.rs

```diff
@@ -0,0 +1,70 @@
+use crate::Error;
+use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
+use std::str::FromStr;
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+/// Opaque type representing allowed licenses for advisory content
+pub enum License {
+    CreativeCommonsZero10,
+    CreativeCommonsCc40,
+}
```

Comment:
This list isn't extensible.

We could add a catch-all variant like `Other(String)`, and then lint that the license is recognized.

Or just use a `String` and use lints to check it's a valid license.

Reply:
isn't there SPDX crate that gives these and non-exhaustive form ?

Reply:
> This list isn't extensible.

Indeed. If I understand correctly we want to be able to parse all correctly formatted advisories (for example to be able to parse future advisories), and do the validation in the linter as much as possible (to validate db with current tooling). I did not identify this at first, I updated the PR with an `Other` field. 

> isn't there SPDX crate we could use that gives these and non-exhaustive form ?

There is `spdx` (by Embark, used in cargo-deny), and I started using it at first. I ended up considering it did not bring much value as we will need to strictly restrict the allowed licenses in a lint, so it was bringing a dependency for nothing.

Reply:
I've hidden the variants to make the underlying type invisible and allow switching to a richer representation later.